Reimplement TO_DATE without a Calendar and fix a lot of bugs an incompatibilities

katzyn opened the pull request:
1. Oracle uses first day of the current month (in current year) as default, not the current month of 1970 that makes no sense.

2. Julian day was calculated incorrectly.

3. `Y`, `YY`, `YYY` patterns in Oracle use current year as a base, so `999` with `YYY` becomes `2999`.

4. 12 AM and 12 PM were broken.

5. `I`, `IY`, `IYY`, `IYYY` are not available in Oracle for this function and should not be availaible, this patterns are not a years, they mean week years. Old implemetation parsed them as normal years.

6. `TO_TIMESTAMP_TZ` from Oracle is implemented. It returns `TIMESTAMP WITH TIME ZONE`.

7. Some tests are changed because they expected the wrong values. All such tests are compared with results from Oracle.

This pull request does not fix all issues with `TO_DATE` family of functions, only some of them that I found.

`Calendar` now used only to get current year and month if these values were not specified.

katzyn committed:
Reimplement TO_DATE without a Calendar and fix a lot of bugs an incompatibilities

katzyn committed:
Remove broken and unexpected support of ISO week years from TO_DATE

grandinj commented on h2/src/main/org/h2/util/ToDateParser.java:
```diff
@@ -32,18 +57,6 @@
      */
     private ToDateParser(ConfigParam functionName, String input, String format) {
         // reset calendar - default oracle behaviour
```
comment now redundant

katzyn committed:
Remove leftover comment

katzyn committed:
Move getTimeZone() to ToChar and change return values to suitable for this class

katzyn commented:
Fixed.

I also moved `getTimeZone()` from `ValueTimestampTimeZone` to `ToChar` and reimplemented it for this class so it returns requested names directly and avoids allocation of unneeded `SimpleTimeZone`.

We don't need this method any more in other places.

katzyn committed:
Pass proper argument to TimeZone.inDaylightTime()

katzyn commented:
Argument passed to `TimeZone.inDaylightTime()` was also wrong, I moved this code to other method but didn't check it.

I just tested behaviour of Oracle and of course it returns `TZD` properly with DST for the specified date, not for the current date. We don't have data type with time zone ID, so it's not easy to write a test for it, because test will depend on system time zone.

Unfortunately, proper test will cause conversion of value to `Calendar`, but this only happens if format contains `TZD` field in it and type is not a `TIMESTAMP WITH TIME ZONE`. `TZR` and regular fields for years, days, etc are not affected, and `TIMESTAMP WITH TIME ZONE` is not affected too.

There is no normal way to check timezone correctly for local date before Java 8, because `java.util.TimeZone` have a lot of bugs in some time zones. `java.util.Calendar` have own bugs, but in the most cases it works more or less properly. It uses internals of `TimeZone` implementations to do its work, not the public API.
